resilience4j-retry/src/main/java/io/github/resilience4j/retry/Retry.java

```diff
@@ -678,8 +678,12 @@ private void onResult(T result) {
             final long delay = retryContext.onResult(result);
 
             if (delay < 1) {
-                promise.complete(result);
-                retryContext.onComplete();
+                try {
+                    retryContext.onComplete();
+                    promise.complete(result);
+                } catch (Exception e) {
+                    promise.completeExceptionally(e);
+                }
```

Comment:
This is the only change introduced by this PR (Regardless of if the new `failAfterMaxAttempts` is set). Let me know if you see any issues with this